api/src/main/java/uk/gov/hmcts/payment/api/model/IdempotencyKeys.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.hmcts.payment.api.model;
+
+import lombok.*;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.Generated;
+import org.hibernate.annotations.GenerationTime;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import javax.persistence.*;
+import java.util.Date;
+
+@Entity
+@Data
+@Builder(builderMethodName = "idempotencyKeysWith")
+@AllArgsConstructor
+@NoArgsConstructor
+@Table(name = "idempotency_keys")
+@IdClass(IdempotencyKeysPK.class)
+public class IdempotencyKeys {
+
+    @Generated(GenerationTime.INSERT)
+    @Column(name = "id", columnDefinition = "serial", updatable = false)
+    private Integer id;
+
+    @Id
+    @Column(name = "idempotency_key")
+    private String idempotencyKey;
```

Comment:
Maybe use UUID instead of string ? Just a thought

Reply:
eventually, we need to use tostring methods to do Db transactions hence better to keep that as it is